tests/conf/i18n/translatehelpervisitor.js

```diff
@@ -0,0 +1,71 @@
+const TranslateHelperVisitor = require('../../../conf/i18n/translatehelpervisitor');
+const Translator = require('../../../conf/i18n/translator');
+const Handlebars = require('handlebars');
+const _ = require('lodash');
+
+async function createTranslator () {
+  const locale = 'fr';
+  return Translator.create(locale, [], { [locale]: { translation: {
+    'Hello': 'Bonjour',
+    'result': 'résultat',
+    'result_plural': 'résultats'
+  } } });
+}
+
+describe('TranslateHelperVisitor usage', () => {
+  it('static translation', async () => {
+    const template = `{{ translate phrase='Hello' }}`;
+    const ast = Handlebars.parse(template);
+    const translator = await createTranslator();
+
+    new TranslateHelperVisitor(translator).accept(ast);
+    const indexOfFirstStatement = 0;
+    const translatedValue = ast.body[indexOfFirstStatement].value;
+
+    expect(translatedValue).toEqual('Bonjour');
+  });
+
+  it('plural translation', async () => {
+    const template = `{{ translate phrase='result' pluralForm='results' count=resultCount }}`;
+    const ast = Handlebars.parse(template);
+    const translator = await createTranslator();
+
+    new TranslateHelperVisitor(translator).accept(ast);
+    const indexOfFirstStatement = 0;
+    const hashPairs = ast.body[indexOfFirstStatement].hash.pairs;
+
+    const expectedPluralForm0 = {
+      key: 'pluralForm0',
+      value: { type: 'StringLiteral', value: 'résultat' }
+    };
+
+    const expectedPluralForm1 = {
+      key: 'pluralForm1',
+      value: { type: 'StringLiteral', value: 'résultats' }
+    };
+
+    const expectedCount = {
+      key: 'count',
+      value: { type: 'PathExpression', parts: ['resultCount'] }
+    };
+
+    const actualPluralForm0 = hashPairs.find(hashPair => hashPair.key === 'pluralForm0');
+    const actualPluralForm1 = hashPairs.find(hashPair => hashPair.key === 'pluralForm1');
+    const actualCount = hashPairs.find(hashPair => hashPair.key === 'count');
+
+    expect(actualPluralForm0).toMatchObject(expectedPluralForm0);
```

Comment:
Could you try passing a the AST to Handlebars.compile and testing that way? I think we could simplify these tests a lot that way

Reply:
Can you explain how that would work?

Reply:
my bad I thought there was an easy way to convert the AST back into the template. I tried some googling and it doesn't look like there are any libraries to do this right now. But this test still feels like it's a little long, maybe split the 3 expects into three different `it` blocks nested inside the plural one?

Reply:
I managed to reduce this quite a bit by passing an array to toMatchObject()